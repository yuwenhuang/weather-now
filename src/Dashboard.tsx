import React, { Component } from 'react';
import styled from 'styled-components'

import City from './components/City';

const DashboardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 20px;
  padding: 16px;
`

const DashboardTitle = styled.h1`
  font-size: 3.5em;
  text-align: center;
  color: ${(props) => props.theme.colors.blackCoffee};
`
const SearchSetionWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 400px;
  background-color: ${(props) => props.theme.colors.middleBlueGreen};
  border-radius: 10px;
  aligh-itens: center;
  justify-content: center;
  margin: 20px;
  padding: 20px;
`;

const CitySectionWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: stretch;
  justify-content: center;
  border-radius: 10px;
  margin: 10px;
  padding: 20px;
`;


const defaultCities = [
  { id: 1, name: 'London'},
  { id: 2, name: 'Berlin'}
]

type CityInfo = {
  id: number,
  name: string
}

interface DashboardState {
  userLocation: string;
  isSubmitted: boolean;
  cities: CityInfo[];
}

class Dashboard extends Component<{}, DashboardState> {
  state = {
    userLocation: "",
    isSubmitted: false,
    cities:[...defaultCities]
  };

  handleSubmit = (event: React.SyntheticEvent) => {
    event.preventDefault();
    this.setState({ isSubmitted: true })
    this.setState({ 
      cities: [...this.state.cities, { id: this.state.cities.length +1, name: this.state.userLocation }]
    })
  }

  handleInputChange = (event: React.FormEvent<HTMLInputElement>): void => {
    this.setState({userLocation: event.currentTarget.value})
  }

  renderUserInput = () => {
    return(
      <form onSubmit={this.handleSubmit}>
        <label>Enter your city:</label>
        <input type="text" value={this.state.userLocation} onChange={this.handleInputChange}></input>
        <input type="submit" value="Submit" />
      </form>)
  }

  renderCities = () => {
    const { cities } = this.state
    
    return(
      <>
      { cities.map( city => {return <City name={city.name} />})} 
      </>
    )
    
  }

  render() { 
    return(
      <React.StrictMode>
        <DashboardWrapper>
          <DashboardTitle>
            Weather Now!
          </DashboardTitle>
          <SearchSetionWrapper>
            {this.renderUserInput()}
          </SearchSetionWrapper> 
          <CitySectionWrapper>
            {this.renderCities()} 
          </CitySectionWrapper>
        </DashboardWrapper>
      </React.StrictMode>
    )  
  }
}

export default Dashboard

