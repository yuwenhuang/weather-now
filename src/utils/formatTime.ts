export const formatTime = (s: number): string => {
    const dtFormat = new Intl.DateTimeFormat('en-GB', {
      timeStyle: 'medium',
      timeZone: 'UTC'
    });
    const result = dtFormat.format(new Date(s * 1e3))
  
    return result
}