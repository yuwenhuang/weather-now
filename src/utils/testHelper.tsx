import React from 'react'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from "react-router-dom"

import { weatherTheme } from '../styles/weatherTheme'

export const wrapper = (child: React.Component) => {
  return( 
  <ThemeProvider theme={weatherTheme}><MemoryRouter>{child}</MemoryRouter></ThemeProvider>
  )
}
