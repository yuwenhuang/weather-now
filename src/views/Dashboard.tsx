import React, { Component } from 'react';
import styled from 'styled-components'

import City from '../components/City';

const DashboardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 20px;
  padding: 16px;
`

const DashboardTitle = styled.h1`
  font-size: 3.5em;
  text-align: center;
`
const SearchSetionWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 400px;
  background-color: ${(props) => props.theme.colors.middleBlueGreen};
  border-radius: 10px;
  aligh-itens: center;
  justify-content: center;
  margin: 20px;
  padding: 20px;
`;

const CitySectionWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: stretch;
  justify-content: center;
  border-radius: 10px;
  margin: 10px;
  padding: 20px;
`;


const defaultCities = [
  { id: 1, name: 'London'},
  { id: 2, name: 'Berlin'}
]

type CityInfo = {
  id: number,
  name: string
}

interface DashboardState {
  userLocation: string;
  isSubmitted: boolean;
  cities: CityInfo[];
}

class Dashboard extends Component<Record<string, never>, DashboardState> {
  state = {
    userLocation: "",
    isSubmitted: false,
    cities: JSON.parse(window.localStorage.getItem("cities") || '0') as CityInfo[]|| defaultCities
  };

  handleSubmit = (event: React.SyntheticEvent):void => {
    event.preventDefault();
    this.setState({ isSubmitted: true })
    const cities = [...this.state.cities, { id: this.state.cities.length +1, name: this.state.userLocation }]
    this.setState({ 
      cities,
    })
    window.localStorage.setItem("cities",JSON.stringify(cities))
  }

  handleInputChange = (event: React.FormEvent<HTMLInputElement>): void => {
    this.setState({userLocation: event.currentTarget.value})
  }

  renderUserInput = (): JSX.Element => {
    return(
      <form onSubmit={this.handleSubmit}>
        <label>Enter your city:</label>
        <input type="text" value={this.state.userLocation} onChange={this.handleInputChange}></input>
        <input type="submit" value="Submit" />
      </form>)
  }

  renderCities = (): JSX.Element => {
    const { cities } = this.state
    
    return(
      <>
        {cities.map( city => {return <City key={city.id} name={city.name} />})} 
      </>
    )
    
  }

  render(): JSX.Element { 
    return(
      <DashboardWrapper>
        <DashboardTitle>
            Weather Now!
        </DashboardTitle>
        <SearchSetionWrapper>
          {this.renderUserInput()}
        </SearchSetionWrapper> 
        <CitySectionWrapper>
          {this.renderCities()} 
        </CitySectionWrapper>
      </DashboardWrapper>
    )  
  }
}

export default Dashboard

