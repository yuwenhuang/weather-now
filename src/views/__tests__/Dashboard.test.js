import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { wrapper } from '../../utils/testHelper'
import Dashboard from '../Dashboard'

it('renders <Dashboard /> component without error', () => {
  const component = renderer.create(
    wrapper(<Dashboard />)
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})