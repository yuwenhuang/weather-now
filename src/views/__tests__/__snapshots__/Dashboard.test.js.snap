// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Dashboard /> component without error 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 20px;
  padding: 16px;
}

.c1 {
  font-size: 3.5em;
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 400px;
  background-color: #99E1D9;
  border-radius: 10px;
  -webkit-aligh-itens: center;
  -webkit-box-aligh-itens: center;
  -ms-flex-aligh-itens: center;
  aligh-itens: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 20px;
  padding: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 10px;
  margin: 10px;
  padding: 20px;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Weather Now!
  </h1>
  <div
    className="c2"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Enter your city:
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
  <div
    className="c3"
  >
    <div>
      Loading...
    </div>
    <div>
      Loading...
    </div>
  </div>
</div>
`;
